spec/rubocop/cop/lint/no_return_in_begin_end_blocks_spec.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Lint::NoReturnInBeginEndBlocks do
+  subject(:cop) { described_class.new(config) }
+
+  let(:config) { RuboCop::Config.new }
+
+  it 'registers an offense when using `#bad_method`' do
+    expect_offense(<<~RUBY)
+      def bad_method
+        @bad_method ||= begin
```

Comment:
I think you should tests as many variations as possible on assignment operators, `=`, `+=` and so on, using `shared_examples`.

Reply:
I've made some changes trying to cover the most common assignment operators. Could you have a look when you have some time, please? Thanks.